### PR TITLE
Fix unchecked merge train wreck

### PR DESCRIPTION
**Pull request by sethrj**
I didn't set up the merge check correctly after #1020 so a broken merge (the first one I merged unfortunately, #1012) got into develop. 

This also moves the building into a separate workflow step and adds a catch-all so that we can more simply require

### CODE REVIEW

File: .github/workflows/build-full.yml [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
 concurrency:
   group: ${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
   cancel-in-progress: true
 
 jobs:
-  build:
-    name: CI
+  # TODO: this currently includes non-GPU builds as well
+  # and won't actually run on device
+  gpu:
```
Comment (pcanal) [inline .github/workflows/build-full.yml:13]: Why name it gpu if it will also run on cpu?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: pr
+
+on:
+  push:
+    branches:
+      - develop
+      - backports/**
```
Comment (pcanal) [inline .github/workflows/pr.yml]: Did you mean to keep that here rather than in its own file?

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+name: nightly
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  #schedule:
+  #  - cron:  '0 1 * * *'
```
Comment (pcanal) [inline .github/workflows/nightly.yml]: You may want to remove this file for now.  It leads to:
```
Error: .github#L1
No jobs defined in `jobs`
```

**Review by pcanal — CHANGES_REQUESTED** [review]
The file nightly.yml as is lead to "check failures"

File: .github/workflows/build-full.yml (hunk 1 above)
Comment (sethrj) [reply .github/workflows/build-full.yml:13]: Once I have time I plan to only use these images for GPU builds, and have separate ones for cpu only. I think. Either way, they're both using images designed for nvidia-docker and rocm-docker.

Comment (pcanal) [reply .github/workflows/build-full.yml:13]: > Once I have time I plan to only use these images for GPU builds

Would it make sense to change the name when that happens?

Comment (sethrj) [reply .github/workflows/build-full.yml:13]: Right now they are GPU builds. We're just not running them on GPU.

**Review by pcanal — APPROVED** [review]
Assuming the test passes, this looks good to go.

**Comment by sethrj** [comment]
@pcanal Have any suggestions (in follow-up) for [this error](https://github.com/celeritas-project/celeritas/actions/runs/6936697810)?
```
Canceling since a deadlock for concurrency group 'refs/heads/develop-1-push' was detected between 'top level workflow' and 'build'
```

**Comment by pcanal** [comment]
not yet :(

**Comment by pcanal** [comment]
I would remove the concurrency group in the inner workflow.  I suspect this is related to: https://github.com/github/vscode-github-actions/issues/135  even-though the group name ought to be different (different workflow name).

**Pull request merged**